source/is-literal.d.ts

```diff
@@ -0,0 +1,247 @@
+import type {Primitive} from './primitive';
+import type {Numeric} from './numeric';
+import type {IsNever, IsNotFalse} from './internal';
+
+/**
+Returns a boolean for whether the given type `T` is the specified `LiteralType`.
```

Comment:
Can you make the first line match the readme description for all the types? We'll eventually have some automation to generate the readme: https://github.com/sindresorhus/type-fest/issues/454

Reply:
`LiteralCheck` is an internal type helper. The public types have matching descriptions, I believe.

I left a comment on the other PR asking how we should word the description: https://github.com/sindresorhus/type-fest/pull/564#issuecomment-1459343557